keystrokes/src/components/Character/CharacterDetails/CharacterDetailPage.js

```diff
@@ -33,13 +33,17 @@ function CharacterDetailPage({ userId, match }) {
 	useEffect(() => {
 		async function collectCharacterInfo() {
 			const path = `/Characters/${characterId}`;
-			const snapShot = await ref.child(path).once('value');
-			if (snapShot.exists()) {
-				const charDetails = snapShot.val();
-				setCharacter(charDetails);
+			try {
+				const snapShot = await ref.child(path).once('value');
```

Comment:
nbd but path could be inline here since it's only used once really
the log below should be a console.error or warn and you could just say failed to collect character ${characterId}